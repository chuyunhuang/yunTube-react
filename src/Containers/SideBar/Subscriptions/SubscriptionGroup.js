import React from 'react';
import './SubscriptionGroup.scss'
import { Subscription } from './Subscription';
import { Divider } from 'semantic-ui-react';
import { SideBarHeader } from '../SideBarHeader/SideBarHeader';

export class SubscriptionGroup extends React.Component {
  render() {
    return (
      <React.Fragment>
        <SideBarHeader title="Subscriptions" />
        <Subscription label="Music Channel" broadcasting />
        <Subscription label="Travel Channel" amountNewVideos={10} />
        <Subscription label="Ted x Talks" amountNewVideos={10} />
        <Divider />

      </React.Fragment>
    )
  }
}
