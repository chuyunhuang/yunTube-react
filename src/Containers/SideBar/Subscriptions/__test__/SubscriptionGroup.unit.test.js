import React from 'react';
import { shallow } from 'enzyme';
import { SubscriptionGroup } from '../SubscriptionGroup';

describe('SubsciptionGroup', () => {
  test('renders an empty SubscriptionGroup', () => {
    const wrapper = shallow(
      <SubscriptionGroup />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('render broadcasting Subscription', () => {
    const wrapper = shallow(
      <SubscriptionGroup broadcasting label="Music Channel" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('renders non-broadcasting Subscription with new videos', () => {
    const wrapper = shallow(
      <SubscriptionGroup amountNewVideos={6} label="Travel Channel" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
