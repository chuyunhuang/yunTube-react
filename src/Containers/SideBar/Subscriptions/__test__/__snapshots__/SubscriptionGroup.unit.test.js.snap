// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubsciptionGroup render broadcasting Subscription 1`] = `
<Fragment>
  <SideBarHeader
    title="Subscriptions"
  />
  <Subscription
    broadcasting={true}
    label="Music Channel"
  />
  <Subscription
    amountNewVideos={10}
    label="Travel Channel"
  />
  <Subscription
    amountNewVideos={10}
    label="Ted x Talks"
  />
  <Divider />
</Fragment>
`;

exports[`SubsciptionGroup renders an empty SubscriptionGroup 1`] = `
<Fragment>
  <SideBarHeader
    title="Subscriptions"
  />
  <Subscription
    broadcasting={true}
    label="Music Channel"
  />
  <Subscription
    amountNewVideos={10}
    label="Travel Channel"
  />
  <Subscription
    amountNewVideos={10}
    label="Ted x Talks"
  />
  <Divider />
</Fragment>
`;

exports[`SubsciptionGroup renders non-broadcasting Subscription with new videos 1`] = `
<Fragment>
  <SideBarHeader
    title="Subscriptions"
  />
  <Subscription
    broadcasting={true}
    label="Music Channel"
  />
  <Subscription
    amountNewVideos={10}
    label="Travel Channel"
  />
  <Subscription
    amountNewVideos={10}
    label="Ted x Talks"
  />
  <Divider />
</Fragment>
`;
